xbmc/video/VideoDatabase.cpp

```diff
@@ -1385,6 +1410,67 @@ int CVideoDatabase::AddToTable(const std::string& table, const std::string& firs
   return -1;
 }
 
+int CVideoDatabase::UpdateRatings(int mediaId, const char *mediaType, const RatingMap& values, const std::string& defaultRating)
+{
+  try
+  {
+    if (NULL == m_pDB.get()) return -1;
+    if (NULL == m_pDS.get()) return -1;
+
+    std::string sql = PrepareSQL("DELETE FROM rating WHERE media_id=%i AND media_type='%s'", mediaId, mediaType);
+    m_pDS->exec(sql);
+
+    return AddRatings(mediaId, mediaType, values, defaultRating);
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "%s unable to update ratings of  (%s)", __FUNCTION__, mediaType);
+  }
+  return -1;
+}
+
+int CVideoDatabase::AddRatings(int mediaId, const char *mediaType, const RatingMap& values, const std::string& defaultRating)
+{
+  int ratingid = -1;
+  try
+  {
+    if (NULL == m_pDB.get()) return -1;
+    if (NULL == m_pDS.get()) return -1;
+
+    for (const auto& i : values)
+    {
+      int id;
+      std::string strSQL = PrepareSQL("SELECT rating_id FROM rating WHERE media_id=%i AND media_type='%s' AND rating_type = '%s'", mediaId, mediaType, i.first.c_str());
+      m_pDS->query(strSQL);
+      if (m_pDS->num_rows() == 0)
+      {
+        m_pDS->close();
+        // doesnt exists, add it
+        strSQL = PrepareSQL("INSERT INTO rating (media_id, media_type, rating_type, rating, votes) VALUES (%i, '%s', '%s', %f, %i)", mediaId, mediaType, i.first.c_str(), i.second.rating, i.second.votes);
+        m_pDS->exec(strSQL);
+        id = (int)m_pDS->lastinsertid();
+      }
+      else
+      {
+        id = m_pDS->fv(1).get_asInt();
+        m_pDS->close();
+        strSQL = PrepareSQL("UPDATE rating SET rating.rating = '%.f', rating.votes = %i WHERE rating.media_id = %i", i.second.rating, i.second.votes, id);
```

Comment:
`id = m_pDS->fv(1).get_asInt();` should be `0` not `1`.

Query should be:
`strSQL = PrepareSQL("UPDATE rating SET rating = '%.f', votes = %i WHERE rating_id = %i", i.second.rating, i.second.votes, id);`

The combination of both typos produces the following errors:
```
00:14:18 2125.637695 T:1724208032   ERROR: SQL: [MyVideos102.db] SQL error or missing database
                                            Query: UPDATE rating SET rating.rating = '0', rating.votes = 0 WHERE rating.media_id = 0
00:14:18 2125.665527 T:1724208032   ERROR: AddRatings (674 - episode) failed
```